lib/r10k/puppetfile.rb

```diff
@@ -212,15 +234,22 @@ def concurrent_accept(visitor, pool_size)
   def modules_queue(visitor)
     Queue.new.tap do |queue|
       visitor.visit(:puppetfile, self) do
-        modules.each { |mod| queue << mod }
+        modules_by_cachedir = modules_by_vcs_cachedir.clone
```

Comment:
I would feel safer if this was addressed before the PR is merged. Just to avoid any potential unforeseen issues.

Reply:
I may be missing something, but I'm not sure how this would matter in `WithModules`. That class doesn't have a `concurrent_accept` mode so far as I can tell, so its modules would be processed first, always in serial, and then the Puppetfile would be processed and this logic would handle the potential conflict.

I can't find any other other places that respect `pool_size` besides what's in the Puppetfile here. What did y'all think was potentially problematic?